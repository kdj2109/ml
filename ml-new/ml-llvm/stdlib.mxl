void printFloatTuples(float[] x,  int len) {
         int i;
         float f;
         for (i = 0; i < len; i = i + 1) {
                 f = $x;
		 printf(f);
		 x = x.+;
         }
}

void printIntTuples(int[] x,  int len) {
         int i;
         int f;
         for (i = 0; i < len; i = i + 1) {
                 f = $x;
		 print(f);
		 x = x.+;
         }
}

void addIntTuples(int[] x, int[] y, int len) {
	/*

	Arguments x and y are integer pointers
	pointing to the first elements in the tuples.
	Length is the length of the tuples being passed in.
	Adds the second tuple to the first tuple, stores in the first
	*/
	int i;

	for (i = 0; i < len; i = i + 1) {
		$x = $x + $y; // $ dereferences the pointer
		x = x.+;      // Incrementing the pointer by 1
		y = y.+;		  // Incrementing the pointer by 1
	}
}
