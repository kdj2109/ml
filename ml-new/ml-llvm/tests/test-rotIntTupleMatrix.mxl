
#include <stdlib.mxl>;

int main(){

    int[3][4:3] a;
    int[3][3:4] b;

    int x;
    int k;
    int l;
    int f;
    int[][][] mat;

    a = [|(3,2,3), (4,5,6), (8,5,6)| (4,3,2),(9,8,7),(3,1,3)|(3,2,1),(4,3,2),(2,1,1)|(6,7,5),(3,2,1),(7,8,9)|];
    
    rotateIntTupMatrixRight(@@@a, @@@b, a.rows, a.columns, a.length);
    mat = @@@b;
    for(x = 0; x<b.rows; x = x+1){
        for( k = 0; k<b.columns; k=k+1){
            for(l=0; l<b.length; l = l+1){
                f = $mat;
                printsl(f);
                mat = mat.+;
            }
        printssl("  ");
        }
    prints(" ");
    }      
    return 0;
}
