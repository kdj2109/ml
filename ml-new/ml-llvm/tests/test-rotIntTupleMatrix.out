675  321  432  323
321  432  987  456
789  211  313  856
